src/ImageSharp/Processing/ProjectiveTransformBuilder.cs

```diff
@@ -300,6 +303,11 @@ public Matrix4x4 BuildMatrix(Rectangle sourceRectangle)
                 matrix *= factory(size);
             }
 
+            if (TransformUtilities.IsNaN(matrix))
```

Comment:
I would also:
- Add this check to `PrependMatrix(...)`  / `AppendMatrix(...)` (better exception locality)
- Add a check to verify if the [determinant](https://docs.microsoft.com/en-us/dotnet/api/system.numerics.matrix4x4.getdeterminant) is a near-zero value. (Both here and to individual append methods.)

Reply:
`PrependMatrix(...) / AppendMatrix(...)` only accept factories. We don't know the value of the matrix until we loop through and apply the functions.

A yes, good catch on the determinant! We can't invert the matrix then! 

How large an epsilon do you suggest?

Reply:
> `PrependMatrix(...) / AppendMatrix(...)` only accept factories.

That's the generic `Prepend` / `Append`. `PrependMatrix()` / `AppendMatrix()` takes a matrix and converts it to a lambda, so we have a chance to validate it's value in those methods.

> How large an epsilon do you suggest?

I think we should use [Constants.EpsilonSquared](https://github.com/SixLabors/ImageSharp/blob/08dcb0707969cdaa957ac8cce696ca4b21ec2def/src/ImageSharp/Common/Constants.cs#L19). Not 100% sure, but it's not critical.

Reply:
Cool. All done